src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/boss_northrend_beasts.cpp

```diff
@@ -321,26 +321,20 @@ class npc_snobold_vassal : public CreatureScript
 
         struct npc_snobold_vassalAI : public ScriptedAI
         {
-            npc_snobold_vassalAI(Creature* creature) : ScriptedAI(creature)
+            npc_snobold_vassalAI(Creature* creature) : ScriptedAI(creature), _instance(creature->GetInstanceScript()), _isActive(false)
             {
-                _instance = creature->GetInstanceScript();
                 _instance->SetData(DATA_SNOBOLD_COUNT, INCREASE);
             }
 
-            void Initialize()
```

Comment:
> as the checker cant tell that Reset is called after create

it's more like as "developers" check the variable in Reset before initializing it, causing "conditional jump on uninitialized value" warning on valgrind, so we chose to always initialize all fields and solve all static analysis issues, even if false positive.